app/uk/gov/hmrc/agentsubscription/service/SubscriptionService.scala

```diff
@@ -84,41 +97,32 @@ class SubscriptionService @Inject() (
       .flatMap { agentRecord =>
         if (agentRecord.isAnASAgent && postcodesMatch(agentRecord.businessPostcode, updateSubscriptionRequest.knownFacts.postcode)) {
           val subscriptionRequest = mergeSubscriptionRequest(updateSubscriptionRequest, agentRecord)
-
-          subscribe(subscriptionRequest, authIds, agentRecord.isAnASAgent, Some(agentRecord.arn))
+          for {
+            arn <- Some(agentRecord.arn) match {
+              case Some(arn) if agentRecord.isAnASAgent => Future.successful(arn)
+              case _ => desConnector.subscribeToAgentServices(subscriptionRequest.utr, desRequest(subscriptionRequest))
+            }
+            updatedAmlsDetails <- agentAssuranceConnector.updateAmls(updateSubscriptionRequest.utr, arn)
```

Comment:
the only difference between the above `createSubscription`  and this `updateSubscription` is  in the 
`createSubscription` we are creating amls record a s a last operation but here in the `updateSubscription` its here - and I cant move this operation further up as we need the `arn`  which apperas to be something wrong here ?